ProteoformSuiteInternal/TdBuReader.cs

```diff
@@ -0,0 +1,222 @@
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using System.IO;
+using DocumentFormat.OpenXml.Packaging;
+using DocumentFormat.OpenXml.Spreadsheet;
+using System.Text.RegularExpressions;
+using UsefulProteomicsDatabases;
+using Proteomics;
+using MzIdentML;
+
+namespace ProteoformSuiteInternal
+{
+    public class TdBuReader
+    {
+
+        //READING IN BOTTOM-UP MORPHEUS FILE
+        public static List<Psm> ReadBUFile(string filename)
+        {
+            List<Psm> psm_list = new List<Psm>();
+            var identifications = new MzidIdentifications(filename);
+            for (int i = 0; i < identifications.Count; i++)
+            {
+                List<Ptm> modifications = new List<Ptm>();
+                for (int p = 0; p < identifications.NumModifications(i); p++)
+                {
+                    ModificationWithMass mod = Lollipop.uniprotModificationTable.Values.SelectMany(m => m).OfType<ModificationWithMass>().Where(m => m.id == identifications.ModificationAcession(i, p)).FirstOrDefault();
+                    if (mod != null) modifications.Add(new Ptm(identifications.ModificationLocation(i, p), mod));
+                    else modifications.Add(new Ptm(identifications.ModificationLocation(i, p), new ModificationWithMass(identifications.ModificationAcession(i, p), null, null, ModificationSites.Any, 0, null, 0, null, null, null)));
+                }
+                psm_list.Add(new Psm(identifications.PeptideSequenceWithoutModifications(i), identifications.StartResidueInProtein(i), identifications.EndResidueInProtein(i), modifications, identifications.Ms2SpectrumID(i), identifications.ProteinAccession(i), identifications.ProteinFullName(i), identifications.ExperimentalMassToCharge(i), identifications.ChargeState(i), (identifications.ExperimentalMassToCharge(i) - identifications.CalculatedMassToCharge(i))));
+            }
+           return psm_list;
+        }
+
+        //Reading in Top-down excel
+        public static List<TopDownHit> ReadTDFile(InputFile file)
+        { 
+                List<TopDownHit> td_hits = new List<TopDownHit>();
+                using (SpreadsheetDocument spreadsheetDocument = SpreadsheetDocument.Open(file.complete_path, false))
+                {
+                    // Get Data in Sheet1 of Excel file
+                    IEnumerable<Sheet> sheetcollection = spreadsheetDocument.WorkbookPart.Workbook.GetFirstChild<Sheets>().Elements<Sheet>(); // Get all sheets in spread sheet document 
+                    WorksheetPart worksheet_1 = (WorksheetPart)spreadsheetDocument.WorkbookPart.GetPartById(sheetcollection.First().Id.Value); // Get sheet1 Part of Spread Sheet Document
+                    SheetData sheet_1 = worksheet_1.Worksheet.Elements<SheetData>().First();
+                    IEnumerable<Row> rowcollection = worksheet_1.Worksheet.Descendants<Row>().ToList();
+                    foreach (Row row in rowcollection)
```

Comment:
This looks similar to ComponentReader. Could you check on whether we could generalize reading an excel file? Might allow us to unit test that Excel file reading is working properly.